clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -6061,6 +6061,40 @@ def CIR_LinkerOptionsOp : CIR_Op<"linker_options", [
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// BlockAddressOp
+//===----------------------------------------------------------------------===//
+
+def CIR_BlockAddressOp : CIR_Op<"blockaddress"> {
+  let summary = "Get the address of a cir.label within a function";
+  let description = [{
+    The `cir.blockaddress` operation takes a label and a function name and
+    produces a pointer value that represents the address of that cir.label within
+    the specified function.
+
+    This operation models GCC's "labels as values" extension (`&&label`), which
+    allows taking the address of a local label and using it as a computed
+    jump target (e.g., with `goto *addr;`).
+
+    Example:
+    ```mlir
+    %addr = cir.blockaddress("label", "foo") -> !cir.ptr<!cir.void>
```

Comment:
Use a FlatSymbolRefAttr for the function name. Function name should come first, label second.